_posts/2018-05-17-running-strimzi-on-openshift-online.md

```diff
@@ -0,0 +1,313 @@
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-job-topic
+  labels:
+    strimzi.io/kind: topic
+    strimzi.io/cluster: my-cluster
+data:
+  name: my-job-topic
+  partitions: "1"
+  replicas: "1"
+  config: |-
+    {
+      "retention.bytes": "1073741824",
+      "retention.ms": "86400000",
+      "segment.bytes": "1073741824"
+    }
+```
+
+This config map will be picked by the Topic Operator, which will create a new topic named "my-job-topic" with one partition and one replica.
+With the topic being ready, we can now deploy the jobs.
+The producer job will send in total 1000 messages at a speed of approximately one message per second.
+After it sends all 1000 messages it will terminate. You can follow the progress in the log.
+The consumer job will be consuming the messages send by the producer job.
+Once it receives all 1000 messages, it will also terminate.
+These jobs will verify that our Kafka deployment really works.
+The YAML below can be used to deploy both jobs.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    app: kafka-producer-job
+  name: kafka-producer-job
+spec:
+  parallelism: 1
+  completions: 1
+  backoffLimit: 5
+  template:
+    metadata:
+      labels:
+        app: kafka-producer
+    spec:
+      containers:
+      - name: kafka-producer
+        image: scholzj/kafka-producer:latest
+        env:
+          - name: BOOTSTRAP_SERVERS
+            value: my-cluster-kafka:9092
+          - name: TOPIC
+            value: my-job-topic
+          - name: TIMER
+            value: "1000"
+          - name: NUMBER_OF_KEYS
+            value: "1"
+          - name: MESSAGE_COUNT
+            value: "1000"
+      restartPolicy: OnFailure
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    app: kafka-consumer-job
+  name: kafka-consumer-job
+spec:
+  parallelism: 1
+  completions: 1
+  backoffLimit: 5
+  template:
+    metadata:
+      labels:
+        app: kafka-consumer
+    spec:
+      containers:
+      - name: kafka-consumer
+        image: scholzj/kafka-consumer:latest
+        env:
+          - name: BOOTSTRAP_SERVERS
+            value: my-cluster-kafka:9092
+          - name: TOPIC
+            value: my-job-topic
+          - name: GROUP_ID
+            value: my-kafka-consumer
+          - name: MESSAGE_COUNT
+            value: "1000"
+      restartPolicy: OnFailure
+```
+
+OpenShift Online is a nice service.
+But the Starter tier is obviously too limited for any serious project - at least one using Apache Kafka.
+We cann't deploy there a proper cluster with multiple nodes and sufficient memory for some real work.
+But if you want to give Strimzi a quick try and do not want to bother with running your own cluster, you can give it a try.
+It will show you the basic principles how the Strimzi operators work.
+If you are on the paid Pro tier you can get lot more resources.
+And if needed, you can deploy bigger cluster together with applications using them.
```

Comment:
"... bigger clusters ... " ? You are referring as "them" at the end of the sentence.